pkg/modulewriter/modulewriter.go

```diff
@@ -90,19 +104,33 @@ func WriteDeployment(blueprint *config.Blueprint, outputDir string, overwriteFla
 		return err
 	}
 
+	metadata := deploymentMetadata{
+		DeploymentMetadata: []groupMetadata{},
+	}
 	for _, grp := range blueprint.DeploymentGroups {
 		writer, ok := kinds[grp.Kind]
 		if !ok {
 			return fmt.Errorf(
 				"Invalid kind in deployment group %s, got '%s'", grp.Name, grp.Kind)
 		}
 
-		err := writer.writeDeploymentGroup(grp, blueprint.Vars, deploymentDir)
+		filteredVars := filterVarsByGraph(blueprint.Vars, grp, graph)
```

Comment:
nit. Filtering appears to be terraform specific logic, it would be reasonable to move it to `tfwritter`.
But we can do it later, once it easier (no explicit graph is passed around).

Reply:
Done, as I agree.